core/app/models/spree/variant.rb

```diff
@@ -418,10 +418,7 @@ def set_master_out_of_stock
 
     # Ensures a new variant takes the product master price when price is not supplied
     def set_price
-      if price.nil? && Spree::Config[:require_master_price] && !is_master?
-        raise 'No master variant found to infer price' unless product && product.master
-        self.price = product.master.price
-      end
+      self.price = product.master.price if price.nil? && Spree::Config[:require_master_price] && !is_master? && product && product.master
```

Comment:
I'm not sure about a guard clause being this long, but it's certainly not blocking.

Reply:
Agreed, it got quite long. Ruby's `statement if condition` is to good to don't use it. :smile:

@kennyadsl contribution might reduce it though - https://github.com/solidusio/solidus/pull/3043#discussion_r249722989.

Reply:
Yes, I agree with his feedback.